### PR TITLE
160 develop TensorBoard event handler

### LINKED ISSUE #160: develop TensorBoard event-handler
**Is your feature request related to a problem? Please describe.**
1. Now we only have a simple example code of TensorBoard logging. All the arguments are fixed and 
    can't be used in other code or user's examples.
2. The TensorBoard example is not correct because:
    (1) It doesn't show loss of every iteration, just show the loss of the last iteration in every epoch.
    (2) It doesn't show any evaluation information, should show 3D output of evaluation instead of train.

**Describe the solution you'd like**
Develop a general ignite Event-handler for TensorBoard, will be easier to use and extend in the future.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #160 .

### Description
This PR implemented an Event handler to support TensorBoard recording.
It can record loss values, metrics and 2D/3D model outputs as images.

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

File: monai/networks/utils.py [hunk 1]
```diff
@@ -52,6 +52,6 @@ def predict_segmentation(logits):
     """
     # generate prediction outputs, logits has shape BCHW[D]
     if logits.shape[1] == 1:
-        return (logits[:, 0] >= 0).int()  # for binary segmentation threshold on channel 0
+        return (logits >= 0).int()  # for binary segmentation threshold on channel 0
     else:
         return logits.max(1)[1]  # take the index of the max value along dimension 1
```
Comment (wyli) [inline monai/networks/utils.py]: line 57 will still remove the channel

File: monai/handlers/tensorboard_handler.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                    '3D images must select only 1 channel to show.'
+                for i in range(len(show_labels)):
+                    img2tensorboard.add_animated_gif(writer, 'output' + str(i), show_outputs[i], 64, 255, current_epoch)
+                    label_tensor = torch.where(show_labels[i] > 0, ones, show_labels[i])
+                    img2tensorboard.add_animated_gif(writer, 'label' + str(i), label_tensor, 64, 255, current_epoch)
+            elif len(show_labels.shape) == 4:
+                for i in range(len(show_labels)):
+                    writer.add_image('output' + str(i), show_outputs[i], current_epoch)
+                    label_tensor = torch.where(show_labels[i] > 0, ones, show_labels[i])
+                    writer.add_image('label' + str(i), label_tensor, current_epoch)
```
Comment (wyli) [inline monai/handlers/tensorboard_handler.py]: should we raise an error when the shape is not 4 or 5?

File: monai/handlers/tensorboard_handler.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        Args:
+            engine (ignite.engine): Ignite Engine, it can be a trainer, validator or evaluator.
+            writer (SummaryWriter): TensorBoard writer, created in TensorBoardHandler.
+
+        """
+        current_epoch = self.global_epoch_transform() if self.global_epoch_transform is not None \
+            else engine.state.epoch
+        summary_dict = engine.state.metrics
+        for name, value in summary_dict.items():
+            writer.add_scalar(name, value, current_epoch)
```
Comment (wyli) [inline monai/handlers/tensorboard_handler.py]: i think this curve plotting is a different topic, should happen in a separate handler. also, shall we support plotting a distribution of the network parameters?

File: monai/networks/utils.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/networks/utils.py]: 1. I think this logic is for "Argmax" but used wrong API (max) here, right?
    I changed to argmax in the latest patch.
2. I think it's still not correct even with argmax, it can support multi-labels segmentation?
Thanks.

File: monai/handlers/tensorboard_handler.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/handlers/tensorboard_handler.py]: fixed in the latest patch.
Thanks.

File: monai/networks/utils.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/networks/utils.py]: Will remove prediction from UNet output in other PR later.

File: monai/handlers/tensorboard_handler.py (hunk 3 above)
Comment (Nic-Ma) [reply monai/handlers/tensorboard_handler.py]: I split it into 2 different tensorboard event handlers in the latest patch.
Thanks.
